modules/snooker/src/main/ts/ephox/snooker/api/TableLookup.ts

```diff
@@ -16,58 +17,37 @@ const lookup = function (tags: string[], element: Element, isRoot: (e: Element)
     return Option.some(element);
   }
 
-  const isRootOrUpperTable = function (elm: Element) {
-    return Selectors.is(elm, 'table') || isRoot(elm);
-  };
+  const isRootOrUpperTable = (elm: Element) => Selectors.is(elm, 'table') || isRoot(elm);
 
   return SelectorFind.ancestor(element, tags.join(','), isRootOrUpperTable);
 };
 
 /*
  * Identify the optional cell that element represents.
  */
-const cell = function (element: Element, isRoot?: (e: Element) => boolean) {
-  return lookup([ 'td', 'th' ], element, isRoot);
-};
+const cell = (element: Element, isRoot?: (e: Element) => boolean) => lookup<HTMLTableCellElement>([ 'td', 'th' ], element, isRoot);
 
-const cells = function (ancestor: Element) {
-  return LayerSelector.firstLayer(ancestor, 'th,td');
-};
+const cells = (ancestor: Element): Element<HTMLTableCellElement>[] => LayerSelector.firstLayer(ancestor, 'th,td');
 
-const notCell = function (element: Element, isRoot?: (e: Element) => boolean) {
-  return lookup([ 'caption', 'tr', 'tbody', 'tfoot', 'thead' ], element, isRoot);
-};
+const notCell = (element: Element, isRoot?: (e: Element) => boolean) => lookup<DomElement>([ 'caption', 'tr', 'tbody', 'tfoot', 'thead' ], element, isRoot);
 
-const neighbours = function (selector: string, element: Element) {
-  return Traverse.parent(element).map(function (parent) {
-    return SelectorFilter.children(parent, selector);
-  });
-};
+const neighbours = (selector: string, element: Element) =>
+  Traverse.parent(element).map((parent) => SelectorFilter.children(parent, selector));
 
 const neighbourCells = Fun.curry(neighbours, 'th,td');
```

Comment:
This won't have the relevant types when it probably should. So I'd suggest changing these to something like this:
```
const neighbourCells: (element: Element) => HTMLTableCellElement[] = Fun.curry(neighbours, 'th,td');
const neighbourRows: (element: Element) => HTMLTableRowElement[]  = Fun.curry(neighbours, 'tr');
```

Reply:
Thanks for the suggestion. Although, it was complaining about incompatible types so I was thinking it might it easier to remove the use of `Fun.curry` and change the `neighbours` function which is fine as it is not exported.